### ISSUE TITLE
New cop: `Lint/UselessMethodDefinition`

### ISSUE THREAD

**Issue by fatkodima** [rubocop-hq/rubocop#8472]
There are at least 2 cases:
1. empty constructors - this may be just overriding parent's constructor, but this is bad anyway 
2. methods just calling `super`

**Comment by ashmaroli** [comment]
I can think of one more use-case:
- *private* methods not used in any of the other instance methods of the Class. Modules shouldn't be affected by the cop since the method may have significance when the module is mixed-in somewhere.

**Comment by biinari** [comment]
For `private` methods not used within the class, what about a child class that uses a `private` method of its parent?

```ruby
class Parent
  private

  def foo
    'something'
  end
end

class Child < Parent
  def bar
    foo
  end
end
```

I'm not sure I would make a strong case for not detecting that but I do have some code like this. It may at least be worth considering as an unsafe autocorrection.
